cleangoldenami/main.go

```diff
@@ -115,26 +183,69 @@ func getPublicImages(ec2Client DescribeImagesClient) ([]ec2Types.Image, error) {
 	return describeImagesResponse.Images, nil
 }
 
-func getOldestImage(images []ec2Types.Image) (ec2Types.Image, error) {
+func hasMatchingTag(image ec2Types.Image, tag string) bool {
```

Comment:
nitpick - can we either

a.) name this `hasVersionTag`
b.) have it take a `key` as well as the value of the tag and use that